torch/onnx/utils.py

```diff
@@ -860,6 +860,12 @@ def _run_symbolic_function(g, n, inputs, env, operator_export_type=OperatorExpor
                 new_node = new_op_outputs[0].node() if n.outputsSize() > 1 else new_op_outputs.node()
                 for b in n.blocks():
                     new_block = new_node.addBlock()
+                    # Copy input metadata to subblock
+                    # This is for Loop only, since If only has a single input.
+                    for i, b_in in enumerate(b.inputs()):
+                        if i < 1:
+                            continue
+                        b_in.setType(n.inputsAt(i + 1).type())
```

Comment:
Is there a chance of going beyond input size with the `i + 1` indexing here?

Reply:
for valid nodes no. Thus updated code to avoid unexpected out of range access.